File: src/main/java/io/github/takenoha/towerdefense/domain/EnemyPathContext.java
package io.github.takenoha.towerdefense.domain;

/** Paper-independent facts supplied to the role-aware navigation planner. */
public record EnemyPathContext(
        boolean directPathAvailable,
        boolean protectedObstacle,
        boolean breakableObstacle,
        boolean buildableGap,
        int consecutivePathFailures) {
    public EnemyPathContext {
        if (consecutivePathFailures < 0) {
            throw new IllegalArgumentException("consecutivePathFailures must not be negative");
        }
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/domain/EnemyPathContext.kt
package io.github.takenoha.towerdefense.domain

import kotlin.jvm.JvmRecord

/** Paper-independent facts supplied to the role-aware navigation planner. */
@JvmRecord
data class EnemyPathContext(
    val directPathAvailable: Boolean,
    val protectedObstacle: Boolean,
    val breakableObstacle: Boolean,
    val buildableGap: Boolean,
    val consecutivePathFailures: Int,
) {
    init {
        if (consecutivePathFailures < 0) {
            throw IllegalArgumentException("consecutivePathFailures must not be negative")
        }
    }
}
